# **Guide to Picking a College Major - Project 2**
### Analyzing the linear relationship between two quantitative variables.

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [7]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image
import statsmodels.api as sm

In [8]:
# Assigns the URL of the image to display to the name 'image_url'.
image_url = 'https://media.istockphoto.com/id/1470208665/photo/multi-ethnic-group-of-latin-and-african-american-college-students-smiling-diversity-portrait.jpg?s=2048x2048&w=is&k=20&c=zicp2F74iFTRKjJUwFBgs_Mb_Xd5vvkvdmYSVoekb1I='

# Display the image
Image(url=image_url, width=600, height=400)


#**Context**

The webiste FiveThirtyEight.com published an article, *The Economic Guide To Picking A College Major*, in 2014 which looked at the earnings of verious college majors. The data analyzed by the author, Ben Casselman, came from the American Community Survey (ACS) 2010-2012 Public Use Microdata Series (PUMS). According to their website, the ACS, part of the U.S. Census Bureau, "is the premier source for detailed population and housing information in our nation."

With this dataset, you have the power to explore college programs and their graduates like never before and create stories of your own!

https://fivethirtyeight.com/features/the-economic-guide-to-picking-a-college-major/

https://www.census.gov/programs-surveys/acs

#**About the Dataset**

This dataset contains 172 rows corresponding to a random sample of people with at least some college education. The data consists of one identifier variable, index, and 22 other variables associated with 172 college majors. Descriptions of the variables are listed below:

**Variables**

| Column name                 | Description                                                |
|-----------------------------|------------------------------------------------------------|
| Index                       | A number assigned to an individual                         |
| Major_code                  | The code associated with the major (Integer)               |
| Major                       | The specific major of the field of study (String)          |
| Major_category              | The category of the major (String)                         |
| Grad_total                  | The total number of graduates from the major (Integer)     |
| Grad_sample_size            | The sample size of graduates from the major (Integer)      |
| Grad_employed               | The number of graduates employed (Integer)                 |
| Grad_full_time_year_round   | The number of graduates employed full-time year-round (Integer) |
| Grad_unemployed             | The number of graduates unemployed (Integer)               |
| Grad_unemployment_rate      | The unemployment rate of graduates (Float)                 |
| Grad_median                 | The median salary of graduates (Integer)                   |
| Grad_P25                    | The 25th percentile salary of graduates (Integer)          |
| Grad_P75                    | The 75th percentile salary of graduates (Integer)          |
| Nongrad_total               | The total number of non-graduates from the major (Integer) |
| Nongrad_employed            | The number of non-graduates employed (Integer)             |
| Nongrad_full_time_year_round| The number of non-graduates employed full-time year-round (Integer) |
| Nongrad_unemployed          | The number of non-graduates unemployed (Integer)           |
| Nongrad_unemployment_rate   | The unemployment rate of non-graduates (Float)             |
| Nongrad_median              | The median salary of non-graduates (Integer)               |
| Nongrad_P25                 | The 25th percentile salary of non-graduates (Integer)      |
| Nongrad_P75                 | The 75th percentile salary of non-graduates (Integer)      |
| Grad_share                  | The share of graduates in the major (Float)                |
| Grad_premium                | The difference between the median salary of graduates and non-graduates (Integer) |


# **A Snippet of the Data**

Let's take a look at the data. To do this, first we import it directly from the url.

In [9]:
# Assigns the URL containing the data file to 'file_path'.
file_path = "https://raw.githubusercontent.com/fivethirtyeight/data/master/college-majors/grad-students.csv"

# Reads in the CSV data file and assigns it to the DataFrame 'df'.
df = pd.read_csv(file_path)


Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [10]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)

# When you type the object name, the object gets printed.
df

Major_code                                       Major  \
0          5601                       CONSTRUCTION SERVICES   
1          6004           COMMERCIAL ART AND GRAPHIC DESIGN   
2          6211                      HOSPITALITY MANAGEMENT   
3          2201      COSMETOLOGY SERVICES AND CULINARY ARTS   
4          2001                  COMMUNICATION TECHNOLOGIES   
..          ...                                         ...   
168        5203                       COUNSELING PSYCHOLOGY   
169        5202                         CLINICAL PSYCHOLOGY   
170        6106     HEALTH AND MEDICAL PREPARATORY PROGRAMS   
171        2303                   SCHOOL STUDENT COUNSELING   
172        2301  EDUCATIONAL ADMINISTRATION AND SUPERVISION   

                          Major_category  Grad_total  Grad_sample_size  \
0    Industrial Arts & Consumer Services        9173               200   
1                                   Arts       53864               882   
2                               Business       24417               437   
3    Industrial Arts & Consumer Services        5411                72   
4                Computers & Mathematics        9109               171   
..                                   ...         ...               ...   
168             Psychology & Social Work       51812               724   
169             Psychology & Social Work       22716               355   
170                               Health      114971              1766   
171                            Education       19841               260   
172                            Education       54159               841   

     Grad_employed  Grad_full_time_year_round  Grad_unemployed  \
0             7098                       6511              681   
1            40492                      29553             2482   
2            18368                      14784             1465   
3             3590                       2701              316   
4             7512                       5622              466   
..             ...                        ...              ...   
168          38468                      28808             1420   
169          16612                      12022              782   
170          78132                      58825             1732   
171          11313                       8130              613   
172          34142                      26850              582   

     Grad_unemployment_rate  Grad_median  Grad_P25  Grad_P75  Nongrad_total  \
0                  0.087543      75000.0     53000  110000.0          86062   
1                  0.057756      60000.0     40000   89000.0         461977   
2                  0.073867      65000.0     45000  100000.0         179335   
3                  0.080901      47000.0     24500   85000.0          37575   
4                  0.058411      57000.0     40600   83700.0          53819   
..                      ...          ...       ...       ...            ...   
168                0.035600      50000.0     36000   65000.0          16781   
169                0.044958      70000.0     47000   95000.0           6519   
170                0.021687     135000.0     70000  294000.0          26320   
171                0.051400      56000.0     42000   70000.0           2232   
172                0.016761      65000.0     50000   86000.0           4003   

     Nongrad_employed  Nongrad_full_time_year_round  Nongrad_unemployed  \
0               73607                         62435                3928   
1              347166                        250596               25484   
2              145597                        113579                7409   
3               29738                         23249                1661   
4               43163                         34231                3389   
..                ...                           ...                 ...   
168             12377                          8502                 835   
169              

# **INSTRUCTIONS**

* Use Python to analyze the data set and complete each of the following.
* Replace ellipsis (...) with the relavent names or code.  
* For problems that require a written response, replace the ellipsis (...) by double clicking the text box to start typing.
* Reference the tutorial from activity for assistance.
* Attend office hours if you still need help.

This assignment is intended to explore the univariate linear relationships between quantitative variables in the data. Choose the dependent and independent variables based on your last name.

| Last Name | Dependent Variable (y) | Independent Variable (x) |
|-----------|------------------------|--------------------------|
| A-L       | Grad_median            | Grad_unemployment_rate   |
| M-Z       | Nongrad_median         | Nongrad_unemployment_rate|




Based on your last name, which variable is explanatory and which is the response?

Explanatory: ...

Response: ...

What do you think the direction (positive/negative) and strength (weak/moderate/strong) will be?

# ...

# **QUESTION 1**

## **Scatterplot**

**1.1)** Construct a scatterplot that could be used to show the relationship between the variables stated above.

In [11]:
# Create a scatter plot using Plotly Express

# Scatter plot for A-L
# STUDENTS: replace ... as stated
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Grad_unemployment_rate',
                 # Replace ... with the response variable name
                 y='Grad_median',
                         # Replace ... with a better x-axis label
                 labels={'Grad_unemployment_rate': 'Graduate Unemployment Rate',
                         # Replace ... with a better y-axis label
                         'Grad_median': 'Graduate Median Salary'})

# Updating layout: Students do not change anything in this block of code.
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot: Replace ... with the name of the scatterplot object (above)
scatter_plot_AL.show()

# Scatter plot for M-Z
# Create a scatter plot using Plotly Express
scatter_plot_MZ = px.scatter(df,
                 x='Nongrad_unemployment_rate', #explanatory variable name
                 y='Nongrad_median', #response variable name
                 labels={'Nongrad_unemployment_rate': 'Non-graduate Unemployment Rate', #updates axis labels
                         'Nongrad_median': 'Non-graduate Median Salary'})

# Updating layout: Students do not change anything in this block of code.
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot: Replace ... with the name of the scatterplot object (above)
scatter_plot_MZ.show()

**1.2)** Describe the relationship between the two variables and include context, and identify the direction, strength, and form.

**A-L:** There is a negative, weak/moderate, linear relationship between graduate unemployment rate and graduate median salary.

**M-Z:** There is a negative, weak/moderate, linear relationship between nongraduate unemployment rate and nongraduate median salary.

Is this what you expected?

Answers will vary.

# **Question 2**
## Correlation Coefficient

**2.1)** Calculate the value of the correlation coefficient.

In [7]:
# Calculate the correlation coefficient

# STUDENTS: Replace ... as stated
# Replace the 1st ... with the response variable name.
# Replace the 2nd ... with the explanatory variable name
correlation_AL = df['Grad_median'].corr(df['Grad_unemployment_rate'])
correlation_MZ = df['Nongrad_median'].corr(df['Nongrad_unemployment_rate'])

# Print the correlation coefficient
print(f"A-L: Correlation coefficient between graduate median salary and unemployment rate: r = {correlation_AL}")
print(f"\nM-Z: Correlation coefficient between nongraduate median salary and unemployment rate: r = {correlation_MZ}")

A-L: Correlation coefficient between graduate median salary and unemployment rate: r = -0.1388560326065584

M-Z: Correlation coefficient between nongraduate median salary and unemployment rate: r = -0.30491531415992135


**2.2)** Interpret the correlation coefficient.

**A-L:** There is a negative, weak, linear relationship between graduate unemployment rate and graduate median salary.

**M-Z:** There is a negative, weak, linear relationship between nongraduate unemployment rate and nongraduate median salary.

# **Question 3**

## Least Squares Regression

**3.1)** Calculate the linear model and add the regression line to the scatterplot.


In [12]:
# Calculate the LSRL and add the regression line to the scatterplot.

# A-L
# Extract the variables from the dataframe
# STUDENTS: Replace ... as stated below.
# Replace ... with the explantory variable name
X = df['Grad_unemployment_rate']
# Replace ... with the response variable name
Y = df['Grad_median']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
# STUDENTS: Do not change anything in this block of code.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
# STUDENTS: Do not change anything in this block of code.
model_AL = sm.OLS(Y, X_with_const).fit()

# If we repeatedly run this section of code then a new regression line is
# added each run, unless we recreate the scatterplot each iteration.
# STUDENTS: Replace ... as stated below.
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Grad_unemployment_rate',
                 # Replace ... with the response variable name
                 y='Grad_median',
                         # Replace ... with a better x-axis label
                 labels={'Grad_unemployment_rate': 'Graduate Unemployment Rate',
                         # Replace ... with a better y-axis label
                         'Grad_median': 'Graduate Median Salary'})

# Updating layout
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_AL.add_scatter(x=X, y=model_AL.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_AL.show()

# M-Z
# Extract the variables from the dataframe
X = df['Nongrad_unemployment_rate']
Y = df['Nongrad_median']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
model_MZ = sm.OLS(Y, X_with_const).fit()

# If we repeatedly run this section of code then a new regression line is
# added each run, unless we recreate the scatterplot each iteration.
scatter_plot_MZ = px.scatter(df,
                 x='Nongrad_unemployment_rate', #explanatory variable name
                 y='Nongrad_median', #response variable name
                 labels={'Nongrad_unemployment_rate': 'Non-graduate Unemployment Rate', #updates axis labels
                         'Nongrad_median': 'Non-graduate Median Salary'})

# Updating layout: Students do not change anything in this block of code.
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_MZ.add_scatter(x=X, y=model_MZ.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_MZ.show()

**3.2)** Print the coefficients (Intercept and Slope)



In [9]:
# Print the coefficients (Intercept and Slope)

# STUDENTS: Do not change the code below.
# The coefficients were calculated when the regression line was added to the
# scatterplot and stored in 'model'. This assigns the values of the y-intercept
# and slope to 'coefficients', then prints the values.

# A-L
coefficients_AL = model_AL.params
coefficients_AL.name='Coefficients'
coefficients_AL

# M-Z
coefficients_MZ = model_MZ.params
coefficients_MZ.name='Coefficients'
coefficients_MZ

# Output the DataFrames
print("A-L Group:")
print(coefficients_AL)
print("\nM-Z Group:")
print(coefficients_MZ)

A-L Group:
const                      81599.071787
Grad_unemployment_rate   -123103.883746
Name: Coefficients, dtype: float64

M-Z Group:
const                         71373.189092
Nongrad_unemployment_rate   -237071.201145
Name: Coefficients, dtype: float64


**3.3)** Write the equation of the line that predicts waist circumference from BMI. To type 𝑦̂ you can type “y-hat” or “predicted y”. Do NOT round.



**A-L:** y-hat = 81599.071787 - 123103.883746x

**M-Z:** y-hat = 71373.189092 - 237071.201145x

# **Question 4**
### Slope
Interpret the slope


**A-L:** We expect the graduate median salary to decrease $123,103.884 for each 1 increase in graduate unemployment rate.

**M-Z:** We expect the nongraduate median salary to decrease $237,071.201 for each 1 increase in nongraduate unemployment rate.

##**QUESTION 5**

Interpret the y-intercept of the least squares regression line in the context of this study. State whether the interpretation is reasonable.

**A-L:** We expect the graduate median salary of $81,599.072 when the graduate unemployment rate is 0.

Yes, this makes sense. (No explanation required)

**M-Z:** We expect the non-graduate median salary of $71,373.189 when the nongraduate unemployment rate is 0.

Yes, this makes sense. (No explanation required)

# **Question 6**
### Prediction

The values in the table below show the median salary for graduates (A-L) and non-graduates (M-Z) for Criminology majors.

Show work to predict the value of your dependent variable for Criminology majors using the actual value of the independent variable shown in the table below. Round to two (2) decimal places.

| Last Name      | Independent Variable (x)                |
|----------------|----------------------------------------|
| **A-L**        | `Grad_unemployment_rate` = 0.045299817 |
| **M-Z**        | `Nongrad_unemployment_rate` = 0.056334794 |


**A-L:** y-hat = 81599.071787 - 123103.883746(0.045299817) = $76,022.49

**M-Z:** y-hat = 71373.189092 - 237071.201145(0.056334794) = $60,633.91

In [33]:
# This is NOT in the student notebook.
# A-L prediction
intercept_AL = model_AL.params['const']
slope_AL = model_AL.params['Grad_unemployment_rate']
predicted_AL = (intercept_AL + slope_AL * 0.045299817)
print(f"A-L: y-hat = {intercept_AL} + {slope_AL}(0.045299817) = {predicted_AL}")

# M-Z prediction
intercept_MZ = model_MZ.params['const']
slope_MZ = model_MZ.params['Nongrad_unemployment_rate']
predicted_MZ = (intercept_MZ + slope_MZ * 0.045299817)
print(f"M-Z: y-hat = {intercept_MZ} + {slope_MZ}(0.056334794) = {predicted_MZ}")


A-L: y-hat = 81599.0717874728 + -123103.88374571885(0.045299817) = 76022.48838180247
M-Z: y-hat = 71373.18909183018 + -237071.20114516764(0.056334794) = 60633.9070639839


# **Question 7**
### Residual
**7.1)** Look up the actual value of your dependent variable for the value of the independent variable above. We know it is for Criminology majors, but be sure to type in the value of the independent variable listed in Question 6 in the code below.

In [20]:
# Filter the datafram to find the row where your independent variable
#   equals the value from the table in Question 6.

#A-L
# STUDENTS:
# Replace the 1st ... with the explantory variable name
# Replace the 2nd ... with the value from the table in Question 6
# Replace the 3rd ... with the response variable name
actual_dependent_AL = df[df['Grad_unemployment_rate']== 0.045299817]['Grad_median']
actual_AL = int(actual_dependent_AL.values[0])
print(f"A-L: The actual value of the dependent variable is: ${actual_AL}")

#M-Z
actual_dependent_MZ = df[df['Nongrad_unemployment_rate']== 0.056334794]['Nongrad_median']
actual_MZ = int(actual_dependent_MZ.values[0])
print(f"M-Z: The actual value of the dependent variable is: ${actual_MZ}")

A-L: The actual value of the dependent variable is: $65000
M-Z: The actual value of the dependent variable is: $50000


**7.2)** Compare the predicted value of your dependent variable from Question 6 to the actual value from 7.1. Use the actual name of your dependent variable, as modeled in activity.


**A-L:** The actual graduate median salary is lower than the predicted value.

**M-Z:** The actual nongraduate median salary is lower than the predicted value.

## **QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions. That is, answer only 8a or 8b, but not both.

**8a)** Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

...

--OR--

**8b)** Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

...


<br><br>
### Once you are done and ready to submit, follow the instructions below to save as a PDF and submit to GradeScope.

### Save as PDF
Note 1: You do not have to select Print Preview. You can print directly from the notebook.
Note 2: Image and graph sizes have been set so you should be able to see them correctly without making any changes to the browser width or the layout (portrait vs landscape).
1. Run all code one last time and make sure your graphs can be seen.
2. File -> Print (or ctrl-p/cmnd-p)
3. Change the "Desination" to PDF.
4. Save the PDF, taking note of where it is saved.

### Submit to GradeScope
**Watch the "GradeScope Submission" video for help.**
1. Login to the Canvas course
2. Click on GradeScope in the course navigation.
3. If you see multiple courses in GradeScope, click on the STAT 108 course
4. Click on the name of the assignment that matches your data set
5. Click on "Submit Work", select PDF
6. Select the PDF you just created
7. You need to tell GradeScope which page each problem answer/output is on. You should see a list of problems on the left, and a display of pages (thumbnails) on the right. Assign pages to questions by clicking on the question number on the left, then clicking on all pages that question is on.
8. After ALL questions have been assigned to their respective page(s), click "Submit"

#### **Still need help? Your STAT 108 team is here to help. Take your laptop to office hours.**
